plugins/org.jkiss.dbeaver.launcher/src/org/eclipse/equinox/launcher/Main.java

```diff
@@ -1841,28 +1835,28 @@ private String resolveLocation(String source, String var, String location) {
         return result.replaceFirst("^~", System.getProperty(PROP_USER_HOME));
     }
 
-    private String getProductVersion() {
-        String appVersion = "";
+    private String getProductProperty(String property) {
+        String appProperty = "";
         URL installURL = getInstallLocation();
         if (installURL == null) {
             return null;
         }
 
         File installDir = new File(installURL.getFile());
-        File eclipseProduct = new File(installDir, ".eclipseproduct"); //$NON-NLS-1$
+        File eclipseProduct = new File(installDir, PRODUCT_SITE_MARKER);
         if (eclipseProduct.exists()) {
             Properties props = new Properties();
             try (FileInputStream inStream = new FileInputStream(eclipseProduct)) {
                 props.load(inStream);
-                appVersion = props.getProperty(PRODUCT_SITE_VERSION);
-                if (appVersion == null || appVersion.trim().length() == 0)
-                    appVersion = ""; //$NON-NLS-1$
+                appProperty = props.getProperty(property);
+                if (appProperty == null || appProperty.trim().length() == 0)
```

Comment:
I'd prefer to throw exception here. These are required properties.

Also we can read this file once, as it was done in original launcher in computeDefaultUserAreaLocation. No need to read the same file to read a single property.

Reply:
Reworked a bit